### PR TITLE
rich 13.7.1

### PR DESCRIPTION
**Pull request by psteyer**
rich 13.7.1 

**Destination channel:** defaults

### Links

- [PKG-5199](https://anaconda.atlassian.net/browse/PKG-5199) 
- [Upstream repository](https://github.com/Textualize/rich)
- [Upstream changelog/diff](https://github.com/Textualize/rich/compare/v13.3.5...v13.7.1)

### Explanation of changes:

- removed upperbound pinnings

[PKG-5199]: https://anaconda.atlassian.net/browse/PKG-5199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 2d11b9b8dd03868f09b4fffadc84a6a8cda574e40dc90821bd845720ebb8e89c
+  sha256: 9be308cb1fe2f1f57d67ce99e95af38a1e2bc71ad9813b0e247cf7ffbcc3a432
 
 build:
   number: 1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:13]:
```suggestion
  number: 0
```
Because it's a new version

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,12 +22,12 @@ requirements:
     - python
     - poetry-core 1.5
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:23]: These packages are redundant if the build backend is poetry:
```
    - setuptools
    - wheel
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -22,12 +22,12 @@ requirements:
...
   run:
-    - markdown-it-py >=2.2.0,<3.0.0
-    - pygments >=2.13.0,<3.0.0
+    - markdown-it-py >=2.2.0
+    - pygments >=2.13.0
     - python
-    - typing_extensions >=4.0.0,<5.0.0  # [py<39]
+    - typing_extensions >=4.0.0  # [py<39]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:28]: The upper bound still exists https://github.com/Textualize/rich/blob/v13.7.1/pyproject.toml#L31

File: recipe/meta.yaml [hunk 4]
```diff
@@ -22,12 +22,12 @@ requirements:
...
   run_constrained:
-    - ipywidgets >=7.5.1,<9
+    - ipywidgets >=7.5.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:30]: The upper bound is still relevant https://github.com/Textualize/rich/blob/7f580bdcf07a3b269a0e786b6a3aa9c804f393cf/pyproject.toml#L33

File: recipe/meta.yaml [hunk 5]
```diff
@@ -22,12 +22,12 @@ requirements:
     - python
     - poetry-core 1.5
   run:
-    - markdown-it-py >=2.2.0,<3.0.0
-    - pygments >=2.13.0,<3.0.0
+    - markdown-it-py >=2.2.0
+    - pygments >=2.13.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:26]: `^2.13.0` means `>=2.13.0,<3.0.0`, see poetry's Caret requirements https://python-poetry.org/docs/dependency-specification/#caret-requirements
 https://github.com/Textualize/rich/blob/7f580bdcf07a3b269a0e786b6a3aa9c804f393cf/pyproject.toml#L32

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [inline recipe/meta.yaml:13]:
```suggestion
  number: 0
```

Comment (lorepirri) [reply recipe/meta.yaml:13]: even though at this point I'm not sure about which build number shall we use

**Review by lorepirri — COMMENTED** [review]
tests missing, and Serhii comments

File: recipe/meta.yaml (hunk 5 above)
Comment (lorepirri) [reply recipe/meta.yaml:26]: we discussed this with @psteyer and having a look at recent other discussions about upper bounds, I suggested to remove upper bounds (also above) see `Pinnings` in our knowledge base

File: recipe/meta.yaml (hunk 4 above)
Comment (lorepirri) [reply recipe/meta.yaml:30]: same as https://github.com/AnacondaRecipes/rich-feedstock/pull/7#discussion_r1665396348

File: recipe/meta.yaml (hunk 3 above)
Comment (lorepirri) [reply recipe/meta.yaml:28]: same as https://github.com/AnacondaRecipes/rich-feedstock/pull/7#discussion_r1665396348

File: recipe/meta.yaml (hunk 2 above)
Comment (psteyer) [reply recipe/meta.yaml:21]: removed!

File: recipe/meta.yaml (hunk 5 above)
Comment (psteyer) [reply recipe/meta.yaml:24]: I removed the upper bounds after my conversation with @lorepirri . What is the best way to proceed on this @skupr-anaconda?

Comment (skupr-anaconda) [reply recipe/meta.yaml:24]: I can't agree with all aspects of `Pinnings` in our knowledge base when it's related to upper bounds that were set by maintainers. Often it's a tradition but sometimes it's a warning that things go wrong and the maintainers explicitly say **don't** touch. The major version bump often says that there are breaking changes and without the upper bound all downstream or upstream packages will fail.
It's about of point of view: by the maintainer or by the package builder/distributor of the software.
Generally, I follow the upstream but sometimes I patch version's constraints if I tested that it works fine.

Comment (lorepirri) [reply recipe/meta.yaml:24]: I'm fine with reintroducing the upper constraints.

**Review by lorepirri — APPROVED** [review]

**Review by skupr-anaconda — APPROVED** [review]
LGTM but I've just noticed that home and dev_url should be modified as https://github.com/Textualize/rich because of a redirect

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**